robot_framework/process.py

```diff
@@ -1,8 +1,256 @@
+
+    # If the name wasn't found in a case look it up in cpr
+    if not name:
+        name = nova_cpr.get_address_by_cpr(cpr, nova_access)['name']
+
+    case_party = CaseParty(
+        role="Primær",
+        identification_type="CprNummer",
+        identification=cpr,
+        name=name,
+        uuid=None
+    )
+
+    # Create a new case
+    case = NovaCase(
+        uuid=str(uuid.uuid4()),
+        title="Sygesikring i almindelighed",
+        case_date=datetime.now(),
+        progress_state='Opstaaet',
+        case_parties=[case_party],
+        kle_number="29.03.00",
+        proceeding_facet="G01",
+        sensitivity="Følsomme"
+    )
+    nova_cases.add_case(case, nova_access)
+    return case
+
+
+def attach_email_to_case(document_name: str, email: Email, case: NovaCase, graph_access: GraphAccess, nova_access: NovaAccess):
+    """Upload the email file to Nova as a document.
+
+    Args:
+        document_name: The name of the new document.
+        email: The email object to upload.
+        case: The nova case to attach to.
+        graph_access: The graph access object used to authenticate against Graph.
+        nova_access: The nova access object used to authenticate against Nova.
+    """
+    mime = graph_mail.get_email_as_mime(email, graph_access)
+    doc_uuid = nova_documents.upload_document(mime, f"{document_name}.eml", nova_access)
+
+    doc = Document(
+        uuid=doc_uuid,
+        title=document_name,
+        sensitivity="Følsomme",
+        document_type="Indgående",
+        document_date=email.received_time,
+        approved=True,
+        description="Automatisk journaliseret af robot.",
+    )
+
+    nova_documents.attach_document_to_case(case.uuid, doc, nova_access)
+
+
+def update_or_create_task(case: NovaCase, nova_access: NovaAccess):
+    """If a task on the case already exists the deadline is moved if necessary.
+    If no task exists a new one is created.
+
+    Args:
+        case: The case to update the task on.
+        nova_access: The nova access object used to authenticate.
+    """
+    deadline = datetime.now() + timedelta(days=7)
+
+    # Check if any tasks already exists on the case
+    tasks = nova_tasks.get_tasks(case.uuid, nova_access)
+
+    # Find a non-finished task
+    task = None
+    for t in tasks:
+        if t.status_code != 'F':
+            task = t
+            break
+
+    # If none found create a new one
+    if not task:
+        task = Task(
+            uuid=str(uuid.uuid4()),
+            title="Ny ansøgning",
+            case_worker_uuid=config.CASE_WORKER_UUID,
+            status_code='N',
+            deadline=deadline
+        )
+        nova_tasks.attach_task_to_case(case.uuid, task, nova_access)
+        return
+
+    # If a task already exists and its deadline is later than
+    # the new deadline, update it
+    if task.deadline.date() > deadline.date():
+        task.deadline = deadline
+        nova_tasks.update_task(task, case.uuid, nova_access)
```

Comment:
I would probably place this beneath the for loop that actually finds the task, but thats a minor thing.